src/core/reactor_backend.cc

```diff
@@ -1484,10 +1490,96 @@ class reactor_backend_uring final : public reactor_backend {
         delete pfd;
     }
     virtual future<std::tuple<pollable_fd, socket_address>> accept(pollable_fd_state& listenfd) override {
-        return _r.do_accept(listenfd);
+        if (listenfd.take_speculation(POLLIN | POLLOUT)) {
+            listenfd.maybe_no_more_recv();
+            socket_address sa;
+            auto maybe_fd = listenfd.fd.try_accept(sa, SOCK_CLOEXEC);
```

Comment:
try/catch to avoid leaking an exception to the caller.

Reply:
elsewhere, we use SOCK_NONBLOCK too. Should we keep it? I think so, it's bad is some sockets have the flags and some don't. Also it will break non-uring recvmsg() calls if we speculate EPOLLIN but the speculation turns out wrong.